File: leet_BST_700_bst_search.cpp
class Solution {
public:
    TreeNode* searchBST(TreeNode* node, int val)
    {
        if (!node)              return NULL ;
        if (node->val == val)   return node ;
        if (node->val > val)    return searchBST(node->left, val);
        else                    return searchBST(node->right, val);
    }
}

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */;

File: leet_BST_700_bst_search.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */


var searchBST = function(node, val) {
    if (!node)
        return null
    if (node.val == val)
        return node
    if (node.val > val)
        return searchBST(node.left, val)
    return searchBST(node.right, val)
};
